lib/auth/accountrecovery.go

```diff
@@ -51,21 +51,21 @@ const (
 	approveRecoveryBadAuthnErrMsg = "invalid username, password, or second factor"
 
 	completeRecoveryGenericErrMsg = "unable to recover your account, please contact your system administrator"
+
+	// MaxFailedAttemptsFromStartRecoveryErrMsg is a user friendly error message to try again later.
+	// This error is defined in a variable so that the root caller can determine if an email needs to be sent.
+	MaxFailedAttemptsFromStartRecoveryErrMsg = "you have reached max attempts, please try again later"
```

Comment:
This is alright, but relying on the error message isn't the best. There is a side channel for gRPC errors[1] that we could use to send an unambiguous signal, but to make a better suggestion I would need to understand where you interpret the error and the path the information takes to get there.

(It is likely that there is a change to to trace that we could make to allow that to happen as well.)

[1] https://pkg.go.dev/google.golang.org/genproto/googleapis/rpc/status#Status